src/qibo/models/grover.py

```diff
@@ -0,0 +1,194 @@
+import numpy as np
+from qibo import gates
+from qibo.config import log, raise_error
+from qibo.models.circuit import Circuit
+
+
+class Grover(object):
+    """Model that performs Grover's algorithm.
+
+    # TODO: Add a few more details on Grover's algorithm and/or reference
+    for example `arXiv:quant-ph/9605043 <https://arxiv.org/abs/quant-ph/9605043>`_
```

Comment:
I added an example reference here but can you please add any other paper you think it is relevant, for example for the iterative method. If there is a different paper for the iterative method it can also go to the docstring of the `iterative` method.